### DISCUSSION TITLE
[COPIER] bypass default answers when updating or creating the project

### DISCUSSION THREAD

**Discussion by Mat-moran** [Tecnativa/doodba#363] [Q&A] [answered]
Hi I am trying to bypass the default answers during the project creation with copier (I did not find the way to do it in the creation so I tried updating it). 

I notice that for example if you change the proxy config (for example `domains_prod` and `domains_test`) it does not change in the prod.yml file when update  so I can not start a project without proxy and after updating the `.copier-answers.yml` to change the proxy configuration or for example the backup conf. 

Am I not doing the right approach working with doodba or am I doing something wrong?, what do you think? 

I followed https://github.com/Tecnativa/doodba-copier-template/blob/stable/docs/faq.md#can-i-have-my-own-private-template to update the project.

Thanks in advance! :smile:

**Comment by joao-p-marques** [comment]
Hi @Mat-moran 

The instructions that better apply here are the [update instructions](https://github.com/Tecnativa/doodba-copier-template#getting-updates-for-your-subproject). This update can serve both to **update your project to the latest version** of the template or to **update your current answers**.

So, even if, initially, you did not set `domains_prod` and `domains_test`, you can run `copier update` and then change your answer to those specific questions. You will, then, get a diff on `prod.yaml` and `test.yaml`, according to the new answers.

The link you provided refers to having a private template in parallel to Doodba Copier Template, if you wish that functionality, but that is not necessary for this kind of update.

Hope this is of some help :wink:

**Reply by Mat-moran** [reply]
Good morning @joao-p-marques  
I did it also but there is only a diff in `.copier-anwers.yaml` and not on `prod.yaml` or `test.yaml`.

I did the follow steps:
- `copier copy gh:Tecnativa/doodba-copier-template prueba` answering to `proxy = 1` and `prod_domains = []` 
- `git add .`
- `pre-commit run`
- `git add .`
- ` git commit ...`
- `copier update` with `proxy = 2` and `prod_domains = [[hosts:'www...']]` 

No diff in `prod.yaml` or in `test.yaml` . :cry:  only in `.copier-answers.yaml`

Am I doing something wrong? 
THANKS :smile:

**Reply by Mat-moran** [reply]
OK it seems that there are variables that can not be update with copier like the proxy configuration. 

if I  generate the project with `proxy=2` `domains_prod=[]` and after I do the `copier update` updating the domains this time there is a diff in `prod.yaml` 

Thanks.

**Comment by yajo** [comment, marked as answer]
> after updating the `.copier-answers.yml` to change the proxy configuration or for example the backup conf.

You should never do that.

It seems to be a common problem among new copier users. I just opened https://github.com/copier-org/copier/pull/317 to clarify the problem. You can browse the docs produced from that PR in https://copier--317.org.readthedocs.build/en/317/updating/#never-change-the-answers-file-manually temporarily.

**Reply by Mat-moran** [reply]
Thanks for the clarification @Yajo  I did it like that the first time and it does not produce any diff as you said. 

Now I am doing it throw the command line and it produce the expected changes, except for the case I said before when you first select  a non proxy configuration and after you want to update the proxy config. It will produce  a diff with the domains in `prod.yaml` (if you set `domains_prod` correctly) but it does not add any traefik configuration at `prod.yaml` .

**Reply by yajo** [reply]
Maybe it got broken at some point due to the wrong update. Try it with `copier copy` instead, giving it the same answers and overwritting those files. Let's see if that way, Copier restores the answers-subproject state.

**Reply by Mat-moran** [reply]
No it does not work neither.
